components/chainhook-sdk/src/indexer/stacks/mod.rs

```diff
@@ -562,6 +591,198 @@ pub fn standardize_stacks_microblock_trail(
     Ok(microblocks)
 }
 
+#[cfg(feature = "stacks-signers")]
+pub fn standardize_stacks_marshalled_stackerdb_chunks(
+    indexer_config: &IndexerConfig,
+    marshalled_stackerdb_chunks: JsonValue,
+    chain_ctx: &mut StacksChainContext,
+    ctx: &Context,
+) -> Result<Vec<StacksStackerDbChunk>, String> {
+    let mut stackerdb_chunks: NewStackerDbChunks =
+        serde_json::from_value(marshalled_stackerdb_chunks)
+            .map_err(|e| format!("unable to parse stackerdb chunks {e}"))?;
+    standardize_stacks_stackerdb_chunks(indexer_config, &mut stackerdb_chunks, chain_ctx, ctx)
+}
+
+#[cfg(feature = "stacks-signers")]
+pub fn standardize_stacks_stackerdb_chunks(
+    indexer_config: &IndexerConfig,
+    stackerdb_chunks: &mut NewStackerDbChunks,
+    chain_ctx: &mut StacksChainContext,
+    ctx: &Context,
+) -> Result<Vec<StacksStackerDbChunk>, String> {
+    use stacks_codec::codec::BlockResponse;
+    use stacks_codec::codec::RejectCode;
+    use stacks_codec::codec::SignerMessage;
+    use stacks_codec::codec::ValidateRejectCode;
+
+    let contract_id = &stackerdb_chunks.contract_id.name;
+    let mut parsed_chunks: Vec<StacksStackerDbChunk> = vec![];
+    for slot in stackerdb_chunks.modified_slots.iter() {
+        let data_bytes = match hex::decode(&slot.data) {
+            Ok(bytes) => bytes,
+            Err(e) => return Err(format!("unable to decode signer slot hex data: {e}")),
+        };
+        let signer_message =
+            match SignerMessage::consensus_deserialize(&mut Cursor::new(&data_bytes)) {
+                Ok(message) => message,
+                Err(e) => return Err(format!("unable to deserialize SignerMessage: {e}")),
+            };
+        let message = match signer_message {
+            SignerMessage::BlockProposal(block_proposal) => {
+                StacksSignerMessage::BlockProposal(BlockProposalData {
+                    block: standardize_stacks_nakamoto_block(&block_proposal.block),
```

Comment:
We should also include the `block_hash`, it can be determined by hashing the header fields:

(rust) https://github.com/stacks-network/stacks-core/blob/a2dcd4c3ffdb625a6478bb2c0b23836bc9c72f9f/stackslib/src/chainstate/nakamoto/mod.rs#L764-L795

(js) https://github.com/hirosystems/signer-monitor/blob/d67b153088ffad9ffcd9249301188d86b09dee95/chunk-parser/src/signer-message.ts#L184-L198

The `index_block_hash` can also be included easily:
```ts
function getNakamotoIndexBlockHash(blockHash: string, consensusHash: string): string {
  const hasher = crypto.createHash('sha512-256');
  hasher.update(Buffer.from(blockHash, 'hex'));
  hasher.update(Buffer.from(consensusHash, 'hex'));
  return hasher.digest('hex');
}
```

Reply:
(can be done in another PR)